crates/store/re_datafusion/src/dataframe_query_common.rs

```diff
@@ -190,6 +191,49 @@ impl DataframeQueryTableProvider {
             chunk_request,
         })
     }
+
+    fn column_to_selector(column: &Column) -> Option<ComponentColumnSelector> {
+        ComponentColumnSelector::from_str(column.name()).ok()
+    }
+
+    fn compute_column_is_not_null_filter(
+        filters: &[&Expr],
+    ) -> Result<Vec<Option<ComponentColumnSelector>>, DataFusionError> {
+        filters
+            .iter()
+            .map(|expr| {
+                match expr {
+                    Expr::IsNotNull(inner) => {
+                        if let Expr::Column(col) = inner.as_ref() {
+                            return Ok(Self::column_to_selector(col));
+                        }
+                    }
+                    Expr::Not(inner) => {
+                        if let Expr::IsNull(col_expr) = inner.as_ref() {
+                            if let Expr::Column(col) = col_expr.as_ref() {
+                                return Ok(Self::column_to_selector(col));
+                            }
+                        }
+                    }
+                    Expr::BinaryExpr(binary) => {
+                        if binary.op == Operator::NotEq {
+                            if let (Expr::Column(col), Expr::Literal(sv))
+                            | (Expr::Literal(sv), Expr::Column(col)) =
+                                (binary.left.as_ref(), binary.right.as_ref())
+                            {
+                                if sv.is_null() {
+                                    return Ok(Self::column_to_selector(col));
+                                }
+                            }
+                        }
+                    }
+                    _ => {}
+                }
+
+                Ok(None)
```

Comment:
Nit: this would read nicer with a helper function

```suggestion
                if is_neq_null(expr) {
                    Ok(Self::column_to_selector(col))
                } else {
                    Ok(None)
                }
```

That would also allow us to unit-test `is_neq_null`